### PR TITLE
feat: registration | Add a notification checkbox without functionality

### LINKED ISSUE #629: [FEATURE] Add checkbox/notifications option to user to choose get notified about events and more
### Detailed description

We can add in registration time, or in setting to enable it, like a toggle button
all notifications, event notifications etc so users could decide to receive updates and promotions.

### Context

User need to have an option to decide if they want to be notified so it will not be a spam for them and they will have unsubscribe from notifications option.

### Possible implementation

In registration add checkbox
image
and in settings add toggle buttons

### Additional information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Happyashbunny**
Closes #629 

I was asked to make a notification checkbox in the registration page. I've made the said checkbox, however it lacks functionality. I've commented out the onchange function to be executed on ticking the checkbox and would like to correspond with the collaborator before enabling it. 

Before

Screenshot 2024-02-06 104917

After

Screenshot 2024-02-06 104941

Again it lacks functionality and I would be interested in working on that too. 

[x] - I've made a notification checkbox
[ ] - Its functional

I'd be interested in learning how to make it functional.

### CODE REVIEW

**Comment by kabir0x23** [comment]
@Happyashbunny hi

thanks for creating a pr

can u change text to
upcoming Internships, Events and Newsletter

**Comment by ArkadiK94** [comment]
@Happyashbunny 
In my opinion,  it is better to put those checkboxs one under another. 
In the new checkbox the default option should be checked.

**Comment by Happyashbunny** [comment]
@kabir0x23 @ArkadiK94 Sure no problem I am on it. Thanks for the review

**Comment by Happyashbunny** [comment]
I've made the changes that you guys @kabir0x23 @ArkadiK94  advised me to make

before

Screenshot 2024-02-06 104941

after

Screenshot 2024-02-07 001437


I've added a 'notifications' to the state 'formData' and it initializes with 'true' value upon render and changes to 'false' on unchecking the checkbox. 
I still haven't made any provisions for sending the said notifications. All I've done is that I've made a checkbox that asks the user whether they'd like to receive notifications and upon unchecking it a variable in the state changes to false.  
I've made the checkboxes one below the other as asked. 

Do lemme know if any further changes are to be made.

**Comment by kabir0x23** [comment]
looks nice, can make the checkbox on top instead of center?

Rest looksgood

**Comment by kabir0x23** [comment]
looks nice, can make the checkbox on top instead of center?

Rest looksgood

**Comment by Happyashbunny** [comment]
@kabir0x23 @ArkadiK94 I've aligned the checkbox on top instead of center. 

Before
Screenshot 2024-02-07 001437

After 
Screenshot 2024-02-07 100141

After mobile view
Screenshot 2024-02-07 100200

Lemme know if any further changes or anything else needs to be done

**Review by ArkadiK94 — DISMISSED** [review]
@Happyashbunny Looks awesome, and I liked that you wrote comments in the code. Thanks for contributing.

**Comment by ArkadiK94** [comment]
By the way, we use in the app Redux. 
you wanted to add the functionality of notifications (you started to add some lines of code related to this).
Your goal is to append it to userData obj
```const userData = { email, termsAndConditions };``` 
before we dispatch it
```dispatch(sendEmailCode(userData));```
Then we will handle the backend...

@Happyashbunny Do you want to do it as part of this pr?
if not , I will merge it , and you will create a new one.

**Comment by Happyashbunny** [comment]
@ArkadiK94 I'd like to try it as a part of this PR itself. Thanks for reviewing my code.

File: src/pages/Register.jsx [hunk 1]
```diff
@@ -29,12 +29,15 @@ const Register = ({ authPopup }) => {
         password: "",
         password2: "",
         termsAndConditions: "",
+        // this is intialized with 'true' as the checkbox is checked upon page render
+        notifications: "true",
```
Comment (ArkadiK94) [inline src/pages/Register.jsx:33]: it is a boolean type and not string so you need to change to
`notifications: true`

**Review by ArkadiK94 — CHANGES_REQUESTED** [review]
@Happyashbunny I have one last comment, it is a small one but important.

File: src/pages/Register.jsx (hunk 1 above)
Comment (Happyashbunny) [reply src/pages/Register.jsx]: @ArkadiK94 Silly me! I'm on it.

**Review by ArkadiK94 — APPROVED** [review]

**Pull request merged**